src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Converters/Object/ObjectWithParameterizedConstructorConverter.cs

```diff
@@ -508,6 +504,9 @@ private void BeginRead(ref ReadStack state, ref Utf8JsonReader reader, JsonSeria
 
             state.Current.NumberHandling = jsonParameterInfo?.NumberHandling;
 
+            state.Current.CtorArgumentState.FoundKey = false;
+            state.Current.CtorArgumentState.FoundValue = false;
+
```

Comment:
Had to put it here to avoid the compilation error.

Reply:
I'm assuming the `FoundKey` and `FoundValues` fields are related to KeyValuePair deserialization. If that's the case, can't they just be removed?